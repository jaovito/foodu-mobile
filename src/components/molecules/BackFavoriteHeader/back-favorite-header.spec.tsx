import {render, fireEvent} from '@test-utils';
import {mock} from '../../../test/mocks';
import {BackFavoriteHeader} from '.';

describe('BackFavoriteHeader tests', () => {
  it('should be defined', () => {
    const header = render(
      <BackFavoriteHeader
        onFavorite={mock.foodHeader.onFavorite}
        onGoBack={mock.foodHeader.onGoBack}
      />,
    );

    expect(header).toBeDefined();
  });

  it('should be able to click on go back button', () => {
    const {getByTestId} = render(
      <BackFavoriteHeader
        onFavorite={mock.foodHeader.onFavorite}
        onGoBack={mock.foodHeader.onGoBack}
      />,
    );

    fireEvent.press(getByTestId('BackButtonID'));

    expect(mock.foodHeader.onGoBack).toBeCalled();
  });

  it('should be able to click on favorite button', () => {
    const {getByTestId} = render(
      <BackFavoriteHeader
        onFavorite={mock.foodHeader.onFavorite}
        onGoBack={mock.foodHeader.onGoBack}
      />,
    );

    fireEvent.press(getByTestId('FavoriteButtonID'));

    expect(mock.foodHeader.onFavorite).toBeCalled();
  });

  it('should match with snapshot', () => {
    const header = render(
      <BackFavoriteHeader
        onFavorite={mock.foodHeader.onFavorite}
        onGoBack={mock.foodHeader.onGoBack}
      />,
    ).toJSON();

    expect(header).toMatchSnapshot();
  });
});
