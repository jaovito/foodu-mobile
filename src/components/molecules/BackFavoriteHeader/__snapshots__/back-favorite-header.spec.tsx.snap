// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackFavoriteHeader tests should match with snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 0,
        "paddingLeft": 39,
        "paddingRight": 39,
        "paddingTop": 157,
      },
    ]
  }
>
  <RNGestureHandlerButton
    activeOpacity={0.3}
    borderless={true}
    collapsable={false}
    delayLongPress={600}
    handlerTag={7}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={
      [MockFunction] {
        "calls": [
          [],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={
      {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 39,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 78,
        "justifyContent": "center",
        "opacity": 1,
        "width": 78,
      }
    }
    testID="BackButtonID"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          [
            {
              "color": "#FFFFFF",
              "fontSize": 27,
            },
          ],
          {
            "fontFamily": "Entypo",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ï…‰
    </Text>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    activeOpacity={0.3}
    borderless={true}
    collapsable={false}
    delayLongPress={600}
    handlerTag={8}
    handlerType="NativeViewGestureHandler"
    onActiveStateChange={[Function]}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    onPress={
      [MockFunction] {
        "calls": [
          [],
        ],
        "results": [
          {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    style={
      {
        "alignItems": "center",
        "borderColor": "#FFFFFF",
        "borderRadius": 39,
        "borderStyle": "solid",
        "borderWidth": 1,
        "height": 78,
        "justifyContent": "center",
        "opacity": 1,
        "width": 78,
      }
    }
    testID="FavoriteButtonID"
  >
    <SvgMock
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 27,
          },
        ]
      }
    />
  </RNGestureHandlerButton>
</View>
`;
